nemo/src/io/parser.rs

```diff
@@ -2431,60 +2443,288 @@ mod new {
         IResult,
     };
 
-    macro_rules! tag_token (
-        ($func_name:ident, $tag: expr) => (
-            fn $func_name(tokens: Tokens) -> IResult<Tokens, Tokens> {
-                verify(take(1usize), |t: &Tokens| t.tok[0].kind == $tag)(tokens)
-            }
-        )
-    );
+    fn combine_spans<'a>(span1: Span<'a>, span2: Span<'a>) -> Result<Span<'a>, str_concat::Error> {
+        // SAFETY: The concatenation of strings is okay, because they originated from the same source string.
+        // The raw offset is okay, because the offset of another span is used.
```

Comment:
pretty sure this is unsound, since the requirement that the two spans are from the same allocation is not actually enforced anywhere. As such, `combine_spans` should itself be `unsafe`.